pkgs/applications/misc/xmrig-proxy/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchFromGitHub, cmake, libuv, libmicrohttpd, libuuid
+, donateLevel ? 0
+}:
+
+stdenv.mkDerivation rec {
+  name = "xmrig-proxy-${version}";
+  version = "2.5.2";
+
+  src = fetchFromGitHub {
+    owner = "xmrig";
+    repo = "xmrig-proxy";
+    rev = "v${version}";
+    sha256 = "1x10mrr58lc207zppzkjnhwah83kpxrqpa3idv01lyasv8mfkxzc";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ libuv libmicrohttpd libuuid ];
+
+  postPatch = ''
+    substituteInPlace src/donate.h --replace "kDonateLevel = 2;" "kDonateLevel = ${toString donateLevel};"
```

Comment:
What does this value mean? what means a level of 2 and what means zero?

Reply:
Should be added as a comment.

Reply:
```console
$ xmrig-proxy
unable to open /nix/store/w8rv7q5kcgv3cxrrbdm1xw27rs1zlycb-xmrig-proxy-2.5.2/bin/config.json: no such file or directory
No pool URL supplied. Exiting.
```

Should be maybe changed to just `config.json` since the nix store is readonly.